### DISCUSSION TITLE
Basic Setup Support

### DISCUSSION THREAD

**Discussion by metakai1** [elizaOS/eliza#497] [Q&A]
A thread for novice users on basic setup issues

**Comment by metakai1** [comment]
Added bot for discord but the bot is not coming online on my server.  How to debug this?  I don't know where to start.  I can't find logfiles either, attempting to generate detailed logs...

**Reply by sirkitree** [reply]
Things to check:
- Do you have the bot setup correctly in Discord?
- Is there any console logs (like where you're running the pnpm start command)?

**Reply by metakai1** [reply]
Hi,  the bot is in discord, but it doesn't come online when i start the agent.  

There's no console log error except for this warning:
bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)

**Reply by sirkitree** [reply]
you should be getting a bunch of messages that state things like:

````
Plugins are:  []
 ["◎ sqlite-vec extensions loaded successfully."] 

 ✓ SUCCESS
   SUCCESS 
   Creating runtime for character 
   (name)

 ✓ SUCCESS
   Agent ID 
   2b31b9af-4675-09d1-9d97-7954eb35299d 

 ["✓ Registering action: CONTINUE"]

 ["✓ Registering action: FOLLOW_ROOM"]

 ["✓ Registering action: UNFOLLOW_ROOM"]

 ["✓ Registering action: IGNORE"]

 ["✓ Registering action: NONE"]

 ["✓ Registering action: MUTE_ROOM"]

 ["✓ Registering action: UNMUTE_ROOM"]

 ["✓ Server running at http://localhost:3000/"]

 ["✓ Registering action: JOIN_VOICE"] 

 ["✓ Registering action: LEAVE_VOICE"]

 ["✓ Registering action: SUMMARIZE_CONVERSATION"]

 ["✓ Registering action: CHAT_WITH_ATTACHMENTS"]

 ["✓ Registering action: TRANSCRIBE_MEDIA"]

 ["✓ Registering action: DOWNLOAD_MEDIA"]

 ["◎ Chat started. Type 'exit' to quit."]

You:  ["✓ Logged in as (bot name)"] 

 ["✓ Use this URL to add the bot to your server:"]

 ["✓ https://discord.com/api/oauth2/authorize?client_id=[redacted]&permissions=0&scope=bot%20applications.commands"]

```

**Reply by metakai1** [reply]
hmmm i see the action registrations logs but i don't see this:
You: ["✓ Logged in as (bot name)"]
["✓ Use this URL to add the bot to your server:"]
["✓ https://discord.com/api/oauth2/authorize?client_id=[redacted]&permissions=0&scope=bot%20applications.commands"]

I also added some debug messages in the DiscordClient contstructor but I don't see them. I don't think the constructor is being called.

do i need to add something to the character.json file to enable the Discord client?

**Reply by sirkitree** [reply]
yes, in the clients line

`"clients": ["direct", "discord"],`

**Reply by metakai1** [reply]
thanks for your patience, i should have checked that first but didn't see it mentioned anywhere.  it's working now

**Comment by DCRRE** [comment]
Not sure if anyone have said this yet. But for windows. Pnpm install on powershell and use WSL for pnpm build. Ubuntu distro. Run on windows afterward with pnpm (character etc.. If errors after that, pnpm clean and install on powershell again.

Any code easy debug with editing software with ai assistance such as cursor.

Feel like this should be put on the top more for newer people.

**Reply by sirkitree** [reply]
WSL may be overkill for some. git bash works well: https://www.perplexity.ai/search/install-git-bash-on-windwos-6O2mZyJPQq.zAYi9g64pNQ

`pnpm clean`
`pnpm install`
`pnpm build`
`pnpm start --character=[your.character.json]`

**Reply by Enferlain** [reply]
so you don't need wsl? or are you missing out on functionality by using git bash?

**Comment by PooriyaAbedini** [comment]


**Reply by DCRRE** [reply]
Did you try wsl? I made a comment about it above. Also did you clone or manually download the zip? I cloned the git before and remember some files missing. So i downloaded the zip, and it worked properly. Otherwise, just ensure a proper build from wsl and run pnpm clean/ install from another client if it does not work.

Hope this helps

**Reply by PooriyaAbedini** [reply]
I'm using Ubuntu  ...
I tried both cloning and downloading the zip, and got the same result ...

**Comment by msurfx** [comment]
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- x86 detected
-- Could not find nvcc, please set CUDAToolkit_ROOT.
CMake Error at CMakeLists.txt:67 (message):
  CUDA was not found


-- Configuring incomplete, errors occurred!
ERR! OMG Process terminated: 1

[node-llama-cpp] To resolve errors related to CUDA compilation, see the CUDA guide: https://node-llama-cpp.withcat.ai/guide/CUDA

Getting this message and loading is looping when run pnpm start

**Comment by AgustinRamiroDiaz** [comment]
Is there a way to create characters with `.json` files and use plugins? I've only seen https://youtu.be/6PZVwNTl5hI?si=QuuwZ_fSRLIGx3Jh&t=474 configure plugins and it's a mess around the codebase: he modifies the `.ts` files in order to be able to import the plugins

**Comment by xueyuanhuang** [comment]
hi a quick question: when I run below command line:
```
git clone https://github.com/elizaos/eliza-starter.git
cd eliza-starter
cp .env.example .env
pnpm i && pnpm build && pnpm start
```

and the res show up or called it error: may I ask how to solve this issue....
```
(base) xueyuanhuang@xueyuanhuangs-MacBook-Air eliza-starter % pnpm i
 WARN  GET https://registry.npmjs.org/@remusao%2Fsmaz-decompress error (ECONNRESET). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://registry.npmjs.org/yocto-queue error (ECONNRESET). Will retry in 10 seconds. 2 retries left.
 WARN  19 deprecated subdependencies found: @cliqz/adblocker-content@1.34.0, @cliqz/adblocker-extended-selectors@1.34.0, @cliqz/adblocker-playwright@1.34.0, @cliqz/adblocker@1.34.0, @discordjs/voice@0.17.0, are-we-there-yet@2.0.0, are-we-there-yet@3.0.1, bin-version-check@6.0.0, gauge@3.0.2, gauge@4.0.4, glob@7.2.3, har-validator@5.1.5, inflight@1.0.6, npmlog@5.0.1, npmlog@6.0.2, puppeteer@19.11.1, request@2.88.2, rimraf@3.0.2, uuid@3.4.0
Packages: +1334
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 1436, reused 1335, downloaded 0, added 0, done
node_modules/.pnpm/canvas@2.11.2/node_modules/canvas: Running install script, failed in 2.5s (skipped as optional)
node_modules/.pnpm/puppeteer@19.11.1_bufferutil@4.0.8_typescript@5.6.3_utf-8-validate@5.0.10/node_modules/puppeteer: Running postinstall script, failed in 639ms
.../node_modules/puppeteer postinstall$ node install.js
│ ERROR: Failed to set up Chromium r1108766! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download.
│ Error: Download failed: server returned code 502. URL: https://storage.googleapis.com/chromium-browser-snapshots/Mac/1108766/chrome-mac.zip
│     at /Users/xueyuanhuang/eliza-starter/node_modules/.pnpm/@puppeteer+browsers@0.5.0_typescript@5.6.3/node_modules/@puppeteer/browsers/lib/cjs/httpUtil.js:121…
│     at ClientRequest.requestCallback (/Users/xueyuanhuang/eliza-starter/node_modules/.pnpm/@puppeteer+browsers@0.5.0_typescript@5.6.3/node_modules/@puppeteer/b…
│     at Object.onceWrapper (node:events:628:26)
│     at ClientRequest.emit (node:events:513:28)
│     at HTTPParser.parserOnIncomingClient (node:_http_client:716:27)
│     at HTTPParser.parserOnHeadersComplete (node:_http_common:117:17)
│     at Socket.socketOnData (node:_http_client:558:22)
│     at Socket.emit (node:events:513:28)
│     at addChunk (node:internal/streams/readable:559:12)
│     at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
└─ Failed in 639ms at /Users/xueyuanhuang/eliza-starter/node_modules/.pnpm/puppeteer@19.11.1_bufferutil@4.0.8_typescript@5.6.3_utf-8-validate@5.0.10/node_modules/puppeteer
 ELIFECYCLE  Command failed with exit code 1.
 ```
